### DISCUSSION TITLE
The "Number range" entity

### DISCUSSION THREAD

**Discussion by vwt12eh8** [home-assistant/architecture#818] [Entity Models] [answered]
We propose to add a new "number range" entity (number_range platform).
Think of it as a Number entity with two values, a lower limit and an upper limit.

This would be useful in the following cases

* Specify automatic adjustment ranges for temperature and humidity
* Limiting the range of battery charging and discharging
* Randomly fluctuating the output of a device within a specified range

The climate platform already has the ability to specify a temperature range.
However, with the addition of this entity, it will be possible to use this for other purposes as well.

I believe there is an existing integration that expresses a similar functionality with two NUMBER entities for upper and lower limits.
However, it is not smart to have to manage two entities, and it is also necessary to deal with the case where the upper and lower limits are specified in reverse by the user.

Please, please, please consider this.

**Comment by frenck** [comment]
What I miss in this proposal, is the use cases. Could you provide a couple more? As in the climate is not a valid case, as the climate entity provides that. 

The second, is ok, could be solved by using two number entities, but are not extremely common I guess? The last one is to generic to consider a use case imho.

And which integrations could benefit from it and implement it today?

If we decide to implement a new entity platforms, it must have significant uses.

../Frenck

**Comment by javicalle** [comment]
I believe that this entity could be usefull for most of the min-max configurations. 
I'm thinking on:
* min-max dimming levels for lights
* upper-lower limits in some covers
* some temp/hum sensors with limit alarms (even some electrical/power sensors)

Some of this cases are already defined in the ZHA standard (despite that not all manufacturers would implement they).
If you consider it appropriate, I can look up the definitions and examples in the implementation.

**Reply by frenck** [reply]
> min-max dimming levels for lights
> upper-lower limits in some covers

Should those be possible properties of those platforms instead?

> some temp/hum sensors with limit alarms (even some electrical/power sensors)

That sounds like automations...

**Reply by javicalle** [reply]
> Should those be possible properties of those platforms instead?

Yes, probably. But having an entity of this kind would not be helpfull to manage in a common way for all tha cases?

It is clear that each platform can create its own mechanisms, but that there are a series of behaviors that IMHO would be worth modeling in this new entity:
* lower range < max range
* graphical representation in the UI

For sure, most of the platforms would create their own implementations, but I think that having a base class is essential to facilitate their use in the graphical part or to generate helpers or services to manipulate them

> > some temp/hum sensors with limit alarms (even some electrical/power sensors)
> 
> That sounds like automations...

I was referring to other types of configurations that could use this new type of entities to help manipulate them. 
Some sensors allow configure (by the user) the min/max values at the device will trigger an alarm (send and alarm signal). No need to automate anything, just to configure the device properties.

**Reply by frenck** [reply]
> Yes, probably. But having an entity of this kind would not be helpfull to manage in a common way for all tha cases?

Yeah, sure, sounds good, but at this point, no concrete ones have been provided and no list of integration that could benefit and implement this today has been provided either.

We don't add new entity platforms just because they might be cool, we add those because there is a direct and wider use.

As requested here: <https://github.com/home-assistant/architecture/discussions/818?notification_referrer_id=NT_kwDOAAL6_7E0NTYzMjM1MzkyOjE5NTMyNw#discussioncomment-3828060>

**Reply by javicalle** [reply]
I see your point.

I can say that it will be helpfull in ZHA integration, but I can not give you any example because there is nothing implemented yet. IMHO it can be an egg-chiken question, but I understand that a change of this type is not addressed until we are sure that it is something that will be useful.

**Comment by frenck** [comment, marked as answer]
We have discussed this today with in an architectural meeting, and decided against this proposal.

We feel this one is to mixed up with our current logic and not fitting our current models.

For that reason, I'll go ahead and close this for now.

../Frenckl
